### ISSUE TITLE
Suddenly task list does not refresh properly

### ISSUE THREAD

**Issue by tomas-knap** [odalic/odalic-ui#286]
Testing on top of current odalic-ui and sti commit 51bac510046b17895d58afefa6135d451000cfe1.

Just create new task. It is not depicted in the task list. You have to refresh the browser to see it!

**Comment by tomas-knap** [comment]
Please try to solve asap, otherwise I would have to search (for the demo)  the version when it was working, cherry pick e.g. selection of the subset of the KBs etc. Thanks!

**Comment by brodecva** [comment]
@istvan-satmari Could you look at it plases ASAP?
@tomas-knap Sorry about that, not sure how could I miss that...

**Comment by tomas-knap** [comment]
Still working for 4915f3db9be580ac6a13b2a2a2f1e3b345e840e7

**Comment by brodecva** [comment]
@tomas-knap @istvan-satmari Weird thing is, that it happened to me only once, when adding the first task. Now, when I restarted the server and refresh the client app, it works even in that case...

**Comment by istvan-satmari** [comment]
Seems to be working now, but the bug was difficult to reproduce, so no guarantees.

**Comment by tomas-knap** [comment]
Still not working properly - I was testing current develop branch, cleared cache and after task was created, I had to refresh two or three times task list before it was shown up. But as I mentioned, it never happened to me before - I think it is still working with commit  4915f3db9be580ac6a13b2a2a2f1e3b345e840e7

**Comment by tomas-knap** [comment]
I think it is even worth then before the fix ;-) - I opened task, closed detail and had to again refresh two times.

**Comment by istvan-satmari** [comment]
Yeah, I suppose it's because of the pagination component. The problem is, it should work (from a logical perspective). And if the issue is "sometimes it's working, sometimes not", it should be race-condition-related, which does not appear to be the cause, unfortunately. (Possibly related to inner implementation of Angular's digest cycle; my guess.)

Anyway, this will take a long time to debug (mainly because the issue is very hard to reproduce), so don't expect a fix so soon.

**Comment by tomas-knap** [comment]
It is enough to fix that in iter 5, do not need the fix urgently for the demo ;-)

**Comment by tomas-knap** [comment]
But we have to fix that, that is something the reviewer can easily see.

**Comment by istvan-satmari** [comment]
So, I hope it is solved now.
The problem seems to be quite complex; to my understanding the angular's digest cycle sometimes ends "pematurely" and watcher on the pagination component simply does not get called. And there is not much I can do since calling another digest cycle results in an error (because there is already one in progress).

It seems that firing the event manually (via angular broadcast) does help - this basically forces the pagination component to update anyway. But, again, no guarantees.
